libraries/ESP8266HTTPClient/src/ESP8266HTTPClient.cpp

```diff
@@ -1336,9 +1348,12 @@ int HTTPClient::handleHeaderResponse()
                         break; // We found a match, stop looking
                     }
                 }
+                continue;
             }
 
-            if(headerLine == "") {
+            headerLine.trim(); // remove \r
```

Comment:
What is the reason to move this trim so far below?

Reply:
so that it doesn't run when it isn't needed - for the majority of cases we only need to trim the value, which is done earlier.